ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/util/BlobSidecarPoolImpl.java

```diff
@@ -0,0 +1,369 @@
+      final RequiredBlockRootDroppedSubscriber requiredBlockRootDroppedSubscriber) {
+    requiredBlockRootDroppedSubscribers.subscribe(requiredBlockRootDroppedSubscriber);
+  }
+
+  public synchronized int getTotalBlobSidecars() {
+    return totalBlobSidecars;
+  }
+
+  public synchronized int getTotalBlobSidecarsTrackers() {
+    return blockBlobSidecarsTrackers.size();
+  }
+
+  private BlockBlobSidecarsTracker getOrCreateBlobSidecarsTracker(
+      final SlotAndBlockRoot slotAndBlockRoot, Consumer<BlockBlobSidecarsTracker> onNew) {
+    BlockBlobSidecarsTracker blockBlobSidecarsTracker =
+        blockBlobSidecarsTrackers.get(slotAndBlockRoot.getBlockRoot());
+    if (blockBlobSidecarsTracker == null) {
+      makeRoomForNewTracker();
+      blockBlobSidecarsTracker = trackerFactory.apply(slotAndBlockRoot);
+      blockBlobSidecarsTrackers.put(slotAndBlockRoot.getBlockRoot(), blockBlobSidecarsTracker);
+      onNew.accept(blockBlobSidecarsTracker);
+    }
+    return blockBlobSidecarsTracker;
+  }
+
+  private void makeRoomForNewTracker() {
+    while (blockBlobSidecarsTrackers.size() > (maxTrackers - 1)) {
+      final SlotAndBlockRoot toRemove = orderedBlobSidecarsTrackers.pollFirst();
+      if (toRemove == null) {
+        break;
+      }
+      removeAllForBlock(toRemove);
+    }
+  }
+
+  private void onFirstSeen(final SlotAndBlockRoot slotAndBlockRoot) {
+    final Optional<Duration> fetchDelay = calculateFetchDelay(slotAndBlockRoot);
+
+    if (fetchDelay.isEmpty()) {
+      return;
+    }
+
+    asyncRunner
+        .runAfterDelay(() -> this.fetchMissingContent(slotAndBlockRoot), fetchDelay.get())
+        .ifExceptionGetsHereRaiseABug();
+  }
+
+  @VisibleForTesting
+  Optional<Duration> calculateFetchDelay(final SlotAndBlockRoot slotAndBlockRoot) {
+    final UInt64 slot = slotAndBlockRoot.getSlot();
+
+    if (slot.isLessThan(getCurrentSlot())) {
+      // old slot
+      return Optional.empty();
+    }
+
+    final UInt64 nowMillis = timeProvider.getTimeInMillis();
+    final UInt64 slotStartTimeMillis = secondsToMillis(recentChainData.computeTimeAtSlot(slot));
+    final UInt64 millisPerSlot = secondsToMillis(spec.getSecondsPerSlot(slot));
+    final UInt64 attestationDueMillis = slotStartTimeMillis.plus(millisPerSlot.dividedBy(3));
+
+    if (nowMillis.isGreaterThanOrEqualTo(attestationDueMillis)) {
+      // late block, we already produced attestations on previous head,
+      // so let's wait our target delay before trying to fetch
+      return Optional.of(Duration.ofMillis(TARGET_WAIT_MILLIS.intValue()));
+    }
+
+    final UInt64 upperLimitRelativeToAttDue =
+        attestationDueMillis.minus(MAX_WAIT_RELATIVE_TO_ATT_DUE_MILLIS);
+
+    final UInt64 targetMillis = nowMillis.plus(TARGET_WAIT_MILLIS);
+
+    final UInt64 finalTime =
+        targetMillis.min(upperLimitRelativeToAttDue).max(nowMillis.plus(MIN_WAIT_MILLIS));
+
+    return Optional.of(Duration.ofMillis(finalTime.minus(nowMillis).intValue()));
```

Comment:
good point... I think at the end it may hold even in gnosis timings.. i'll think about it

Reply:
I think for fast slot networks, we would end up being always on the `MIN_WAIT_MILLIS` which makes sense to me.
And I think it is acceptable that value being constant instead of being dependant by slot time, since that number is more related to network latency than actual slot time. We could argue that on fast nets, blocks (and blobs?) could be smaller so the expected latency could be lower on average, but for now seems like we can leave with a constant.